### ISSUE TITLE
TLS resumption

### ISSUE THREAD

**Issue by luengnat** [eclipse-vertx/vert.x#2645]
Since SSL handshake can be quite expensive, it might make sense to have an option to turn on more complex logic in connection pooling for SSL connection to establish at least one connection at the startup so that the TLS session can be cached by the server. This will reduce the CPU time spent by eventloop thread significantly especially when there are multiple requests to the same server at the same time.

**Comment by luengnat** [comment]
@vietj  that's right. Some web browser implements it this way to avoid the expensive to SSL handshake

**Comment by vietj** [comment]
Usually HTTP server tends to make to close unused connections to free up ports, so I've mixed feelings for this.

In 4.0 we plan to provide a connection API for the HttpClient that would allow you to manage yourself HTTP connections manually.

**Comment by hjames9** [comment]
Is this going to be handled in the generic TCP client as well?

**Comment by vietj** [comment]
the generic TCP client does not need to, there is no pooling there so calling NetClient#connect does what you expect.
